src/toil/worker.py

```diff
@@ -259,10 +259,6 @@ def badWorker():
                 else:
                     jobGraph.command = jobGraph.checkpoint
 
-                # Reduce the retry count
-                assert jobGraph.remainingRetryCount >= 0
-                jobGraph.remainingRetryCount = max(0, jobGraph.remainingRetryCount - 1)
```

Comment:
Yes, to make this clean this should really be fixed on startup, plus it keeps the I/O on the leader in one place. Up to you, we can also merge this in now and then do a review later.